src/utils.cpp

```diff
@@ -443,24 +430,38 @@ void getVpdDataInVector(const std::string& vpdFilePath,
     }
 }
 
-size_t getVPDOffset(const nlohmann::json& parsedJson,
+size_t getVPDOffset(const nlohmann::json& i_jsonObj,
                     const std::string& vpdFilePath)
 {
     size_t vpdOffset = 0;
-    if (!vpdFilePath.empty())
+
+    if (vpdFilePath.empty())
+    {
+        throw types::CommonError::InvalidArgument();
+    }
+
+    if (!i_jsonObj.contains("frus"))
+    {
+        throw JsonException("Missing frus section in VPD JSON",
+                            INVENTORY_JSON_SYM_LINK);
+    }
+
+    const nlohmann::json& listOfFrus =
+        i_jsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    for (const auto& itemFRUS : listOfFrus.items())
```

Comment:
Why these calculations should be part of this API. The caller should pass the path for which offset is required.

Reply:
If the format is similar to what we have now in JSON,
Then the caller need to have that intelligence and pass the primary key to this API. The purpose of this API is just to read the offset tag based on the path passed. Having redundant path is not a generic case and is not applicable to all the FRUs.